Error in topologySpreadConstraints for statefulset configuration

### What happened?

I encountered an error while deploying the Redpanda environment using the helm-chart. The error message is "statefulset.topologySpreadConstraints: Invalid type. Expected: object, given: array".

### What did you expect to happen?

I expected the Redpanda environment to be deployed successfully with the defined topology spread constraints in the statefulset configuration.

### How can we reproduce it (as minimally and precisely as possible)?. Please include values file.

To reproduce the error, follow these steps:

Install the Redpanda helm-chart version 4.0.4.
Modify the values.yaml file to include the following statefulset configuration:
```yaml
statefulset:
  replicas: 6                                        
  updateStrategy:
    type: RollingUpdate                              
  budget:
    maxUnavailable: 1                                
  nodeAffinity: 
    preferredDuringSchedulingIgnoredDuringExecution:
      - weight: 1
        preference:
          matchExpressions:
            - key: node-type
              operator: In
              values:
                - primary                            
  topologySpreadConstraints:                         
    - maxSkew: 1                                       
      topologyKey: availability-zone                   
      whenUnsatisfiable: DoNotSchedule                 
      labelSelector:
        matchLabels:                                   
          app.kubernetes.io/instance: redpanda
          app.kubernetes.io/name: redpanda
    - maxSkew: 1                              
      topologyKey: kubernetes.io/hostname      
      whenUnsatisfiable: ScheduleAnyway        
      labelSelector:
        matchLabels:                           
          app.kubernetes.io/instance: redpanda
          app.kubernetes.io/name: redpanda
```
Run the following command to install the Redpanda environment: `helm install redpanda helm-redpanda/redpanda -f statefulset.yaml`
The deployment will fail with the error message:
```console
coalesce.go:220: warning: cannot overwrite table with non table for redpanda.statefulset.topologySpreadConstraints (map[maxSkew:1 topologyKey:topology.kubernetes.io/zone whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for redpanda.statefulset.topologySpreadConstraints (map[maxSkew:1 topologyKey:topology.kubernetes.io/zone whenUnsatisfiable:ScheduleAnyway])
Error: INSTALLATION FAILED: values don't meet the specifications of the schema(s) in the following chart(s):
redpanda:
- statefulset.topologySpreadConstraints: Invalid type. Expected: object, given: array
```

### Anything else we need to know?

I suspect that the issue could be related to the definition of the topologySpreadConstraints in the `template/statefulset.yaml` file of the Redpanda helm-chart. It seems that the definition is not an array. Please check if the topologySpreadConstraints are correctly defined as an array in the template/statefulset.yaml file.

### Which are the affected charts?

Redpanda

### Chart Version(s)

<details>

```console
$ helm -n redpanda list
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS  CHART           APP VERSION
redpanda        redpanda        1               2023-04-27 16:47:47.4519181 +0200 CEST  failed  redpanda-4.0.4  v23.1.7

```

</details>


### Cloud provider

<details>
Private Cloud TANZU Kubernetes
</details>

--- a/charts/redpanda/Chart.yaml
+++ b/charts/redpanda/Chart.yaml
@@ -23,7 +23,7 @@ type: application
 # The chart version and the app version are not the same and will not track
 # together. The chart version is a semver representation of changes to this
 # chart.
-version: 4.0.14
+version: 4.0.15
 
 # The app version is the default version of Redpanda to install.
 appVersion: v23.1.8

--- a/charts/redpanda/templates/statefulset.yaml
+++ b/charts/redpanda/templates/statefulset.yaml
@@ -407,13 +407,13 @@ spec:
 {{- end }}
 {{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion }}
       topologySpreadConstraints:
-      - labelSelector:
-          matchLabels: {{ include "statefulset-pod-labels" . | nindent 12 }}
-  {{- with .Values.statefulset.topologySpreadConstraints }}
-        maxSkew: {{ .maxSkew }}
-        topologyKey: {{ .topologyKey }}
-        whenUnsatisfiable: {{ .whenUnsatisfiable }}
-  {{- end }}
+    {{- range $v := .Values.statefulset.topologySpreadConstraints }}
+        - maxSkew: {{ $v.maxSkew }}
+          topologyKey: {{ $v.topologyKey }}
+          whenUnsatisfiable: {{ $v.whenUnsatisfiable }}
+          labelSelector:
+            matchLabels: {{ include "statefulset-pod-labels" $ | nindent 14 }}
+    {{- end }}
 {{- end }}
 {{- with ( include "statefulset-nodeSelectors" . ) }}
       nodeSelector: {{- . | nindent 8 }}

--- a/charts/redpanda/values.schema.json
+++ b/charts/redpanda/values.schema.json
@@ -635,22 +635,20 @@
           "type": "array"
         },
         "topologySpreadConstraints": {
-          "type": "object",
-          "required": [
-            "maxSkew",
-            "topologyKey",
-            "whenUnsatisfiable"
-          ],
-          "properties": {
-            "maxSkew": {
-              "type": "integer"
-            },
-            "topologyKey": {
-              "type": "string"
-            },
-            "whenUnsatisfiable": {
-              "type": "string",
-              "pattern": "^(ScheduleAnyway|DoNotSchedule)$"
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "properties": {
+              "maxSkew": {
+                "type": "integer"
+              },
+              "topologyKey": {
+                "type": "string"
+              },
+              "whenUnsatisfiable": {
+                "type": "string",
+                "pattern": "^(ScheduleAnyway|DoNotSchedule)$"
+              }
             }
           }
         },

--- a/charts/redpanda/values.yaml
+++ b/charts/redpanda/values.yaml
@@ -530,7 +530,7 @@ statefulset:
   # For details,
   # see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/).
   topologySpreadConstraints:
-    maxSkew: 1
+  - maxSkew: 1
     topologyKey: topology.kubernetes.io/zone
     whenUnsatisfiable: ScheduleAnyway
   securityContext:

